tests/test_engine.py

```diff
@@ -47,6 +47,18 @@ def load_rulebook(rules_file):
     return ruleset_queues, event_log
 
 
+async def get_queue_item(queue, timeout=0.5, times=1):
```

Comment:
Why not making it more time consumption efficient based only on the timeout? something like:
```
async def wait_for_queue_item(queue: asyncio.Queue = None, timeout=0.5):
    start = time.time()
    while True:
        if time.time() - start > timeout:
            return None
        try:
            item = queue.get_nowait()
        except asyncio.QueueEmpty:
            await asyncio.sleep(0.1)
            continue
        if item:
            return item
```

Reply:
@Alex-Izquierdo For the test I am not expecting any activity for 10 seconds because the timer will pop after 10 seconds so doing a sleep for 10 seconds in 0.1 second interval causes unnecessary churn leading to "Ideal Wakeups"